### PR TITLE
await: 1.0.2 -> 1.0.5

### PR DESCRIPTION
**Pull request by chewblacka**
- Added basic version test
- Updated `await` from 1.0.2 to latest version 1.0.5:
https://github.com/slavaGanzin/await/releases

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by zakame** [comment]
Result of `nixpkgs-review pr 349294` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>await</li>
  </ul>
</details>

File: pkgs/by-name/aw/await/package.nix [hunk 1]
```diff
@@ -30,11 +32,18 @@ stdenv.mkDerivation rec {
     install -Dm755 await -t $out/bin
     install -Dm444 LICENSE -t $out/share/licenses/await
     install -Dm444 README.md -t $out/share/doc/await
-    installShellCompletion --cmd await autocomplete.{bash,fish,zsh}
+    installShellCompletion --cmd await \
+      --bash autocompletions/await.bash \
+      --fish autocompletions/await.fish \
+      --zsh autocompletions/await.zsh
```
Comment (NickCao) [inline pkgs/by-name/aw/await/package.nix:38]:
```suggestion
    installShellCompletion --cmd await autocompletions/await.{bash,fish,zsh}
```

Comment (chewblacka) [reply pkgs/by-name/aw/await/package.nix]: Done

**Comment by NickCao** [comment]
The version bump LGTM, for the version tests, we should either use the finalAttrs pattern or versionCheckHook, to ensure the referenced one is actually the one we are building.

File: pkgs/by-name/aw/await/package.nix [hunk 2]
```diff
@@ -3,20 +3,24 @@
...
     repo = "await";
     rev = "v${version}";
-    hash = "sha256-qvSRuRLZnUptXYknyRn4GgmYtj9BnI8flN6EhadbKMw=";
+    hash = "sha256-0U9eLQDvHnRUJt46AI4bDWZfGynqjaWs9teidWP3RsA=";
   };
 
-  nativeBuildInputs = [ installShellFiles ];
+  nativeBuildInputs = [
+    installShellFiles
+    versionCheckHook
```
Comment (NickCao) [inline pkgs/by-name/aw/await/package.nix:22]: `versionCheckHook` should be in `nativeInstallCheckInputs`, please see `doc/hooks/versionCheckHook.section.md`

**Review by NickCao — APPROVED** [review]

**Pull request merged**